frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.unit.spec.tsx

```diff
@@ -0,0 +1,81 @@
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import { renderWithProviders, screen } from "__support__/ui";
+import type { UiParameter } from "metabase-lib/parameters/types";
+import { createMockUiParameter } from "metabase-lib/parameters/mock";
+import ParameterSidebar from "./ParameterSidebar";
+
+interface SetupOpts {
+  initialParameter: UiParameter;
+  otherParameters: UiParameter[];
+}
+
+const setup = ({ initialParameter, otherParameters }: SetupOpts) => {
+  const TestWrapper = () => {
+    const [parameter, setParameter] = React.useState(initialParameter);
+
+    const onChangeName = (newName: string) => {
+      setParameter(prevParameter => ({ ...prevParameter, name: newName }));
+    };
```

Comment:
I figured I needed to create this `TestWrapper` to substitute for the parameter name being updated with redux and prop drilling down ascendant components.